### PR TITLE
Fix the use of time() in measuring performances (replace with time_ns())

### PR DESCRIPTION
**Pull request by nico202**
time() is a wrapper for gettimeofday, which is affected by by discontinuous
jumps in the system time (e.g., if the system administrator manually changes the
system time), see `man gettimeofday`.  On the other hand, time_ns() is a wrapper
for libuv uv_hrtime that has no drift, whose manual page says "The primary use
is for measuring performance between intervals.".

Tentative fix for bug #34115

### CODE REVIEW

**Comment by nico202** [comment]
I'm not on a computer where I can ran tests right now, and I possibly missed some use

File: test/file.jl [hunk 1]
```diff
@@ -382,7 +382,7 @@ if Sys.iswindows()
 else
     @test filesize(dir) > 0
 end
-nowtime = time()
+nowtime = time_ns()
```
Comment (JeffBezanson) [inline test/file.jl]: This doesn't look correct, since this needs to be in the same units as file `mtime`.

**Comment by JeffBezanson** [comment]
Thanks, I agree we should change most of these.

**Review by JeffBezanson — APPROVED** [review]

**Comment by nico202** [comment]
Thanks! Tests are passing (I'm 90% into it, should end in a few minutes). I'll rebase it

**Comment by nico202** [comment]
Here we go :)

**Comment by nico202** [comment]
Is windows failure my fault?

File: stdlib/Distributed/src/cluster.jl [hunk 2]
```diff
@@ -1023,10 +1022,10 @@ function _rmprocs(pids, waitfor)
             end
         end
 
-        start = time()
-        while (time() - start) < waitfor
+        start = time_ns()
+        while (time_ns() - start) < waitfor
```
Comment (JeffBezanson) [inline stdlib/Distributed/src/cluster.jl]:
```suggestion
        while (time_ns() - start) < waitfor*1e9
```

**Comment by nico202** [comment]
If everything is fine I can squash again. Arm 64 error seems unrelated, right?

**Comment by KristofferC** [comment]
The commits can be squashed on merge so no need to do it manually.

**Comment by nico202** [comment]
@KristofferC fine! Didn't know, thanks

**Pull request merged**